<launch>

  <arg name="detector_type" default="ORB" /> # ORB, SURF, or GFT (Good features to track)

  <node pkg="ccny_rgbd" type="vo_icp_node" name="vo_icp_node" 
    output="screen">

    <param name="history_size"      value="5"/>
    <param name="publish_keyframes" value="false"/>

	# Frame names (frame_id)    
    <param name="fixed_frame" value="/odom"/>
    <param name="base_frame" value="/camera_link"/>
       
    <param name="feature/detector_type"       value="$(arg detector_type)"/> #  ORB, SURF, or GFT (Good features to track)
    <param name="feature/smooth"              value="0"/>
    <param name="feature/max_range"           value="7.0"/>

    <param name="reg/max_iterations"          value="25"/>
    <param name="reg/max_corresp_dist"        value="0.15"/>
    <param name="reg/tf_epsilon"              value="1e-15"/>
    <param name="reg/use_ransac_rejection"    value="true"/>
    <param name="reg/ransac_inlier_threshold" value="0.15"/>

  </node>

  <node pkg="ccny_3dmap_server" type="map_server_node" name="map_server" 
    output="screen">

    <!--remap from="/camera/depth_registered/points/" to="/keyframes"/-->

    <param name="fixed_frame"  value="/odom"/>
    <param name="resolution"   value="0.02"/>
    <param name="max_range"    value="5"/>

  </node>

<!--
  <node pkg="ab_filter" type="ab_filter_pose_node" name="ab_filter_pose"
    output="log">

    <remap from="/mav/pose"    to="/rgbd_vo"/>
    <remap from="/mav/pose_f"  to="/rgbd_vo_f"/>
    <remap from="/mav/twist_f" to="/vel"/>
  
    <param name="alpha"   value="1.0"/>
    <param name="beta"    value="0.1"/>

  </node>
-->


</launch>

